src/substrait/common/Decimal.cpp

```diff
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/common/Decimal.h"
+
+namespace io::substrait::common {
+
+Decimal Decimal::fromProto(
+    const ::substrait::proto::Expression_Literal_Decimal& proto) {
+  return {proto.value(), proto.precision(), proto.scale()};
+};
+
+std::string Decimal::toString() {
```

Comment:
With this implementation I'm trying to avoid any normalization and the documented form in #33 matches that.  The counterpoint is text format readability.  How about we leave the simplistic form for now and add making decimal strings more readable a separate task?

Reply:
I was rereading the SQL->Substrait documentation on the website and found that we refer to decimal types as decimal(10, 2) meaning that the precision and scale are considered to be part of the type.  If this is right approach, then what should be here would be 12345_decimal<5,2> (which was what was in the literal docs until I changed it to match here).  I'm leaning towards this although it make it more annoying to specify decimals (and handling decimal as a type will also be more complicated).

Reply:
I've been thinking about how decimal is used and I've decided both the approach taken here and the one in substrait is probably wrong.  The type portion of decimal is essentially the decimal type and its precision (which can be contained within particular implementations of decimal including decimal128 and decimal256 but more on these later).  The scale is just a convenience factor showing where the decimal point is and if you combine two decimals the scale does not need to remain the same.  Thus 123.45_decimal<5> is closer to what we should be using and the substrait definition needs updating as well.

decimal128 and decimal256 are physical versions of the type logical types decimal<38> and decimal<76> (and any precision up through 76 could be handled by decimal256).  I don't see a notion of decimal256 in substrait at this time so sticking to decimal would be fine for now.  However at the point decimal256 is introduced we would likely want to have the specific types as we have i8, i16, and i32 today:
`123.45_decimal128<5>
123.45_decimal256<5>`